### ISSUE TITLE
[BUG] [python] Clamp doesn't work as expected (only clamps red channel)

### ISSUE THREAD

**Issue by polymetric** [OpenImageIO/oiio#3264]
**Describe the bug**
while trying to clamp an image to the 0-1 range using the OIIO python bindings, i discovered that the clamp function behaves quite strangely. i also discovered that the min and max functions are missing from the python bindings for some reason (even though they are present in the pybind headers) I'll open a separate issue for that if i have time

**To Reproduce**
```python
>>> a = oiio.ImageBufAlgo.fill((10, 10, 10), roi=oiio.ROI(0,512,0,512,0,1,0,3))
>>> a.getpixel(0, 0)
(10.0, 10.0, 10.0)
# putting our modified image into a new variable `b` just to ensure it isn't
# some weird memory access thing
>>> b = oiio.ImageBufAlgo.clamp(a, 0, 1)
>>> b.getpixel(0, 0)
(1.0, 10.0, 10.0)
# even with clampalpha, it still doesn't clamp anything but the first channel:
>>> b = oiio.ImageBufAlgo.clamp(a, 0, 1, clampalpha01=True)
>>> b.getpixel(0, 0)
(1.0, 10.0, 10.0)
# just for completeness, it doesn't work on the lower bound either:
>>> a = oiio.ImageBufAlgo.fill((-1, -1, -1), roi=oiio.ROI(0,512,0,512,0,1,0,3))
>>> b = oiio.ImageBufAlgo.clamp(a, 0, 1, clampalpha01=True)
>>> b.getpixel(0, 0)
(0.0, -1.0, -1.0)
# or if i swap the bounds:
>>> b = oiio.ImageBufAlgo.clamp(a, 1, 0, clampalpha01=True)
>>> b.getpixel(0, 0)
(1.0, -1.0, -1.0)
```

**Expected behavior**
Clamps all three channels to specified min or max value, i.e. getpixel returns (1.0, 1.0, 1.0)

**Platform information:**
 - OIIO branch/version: 2.0.5~dfsg0-1
 - OS: Debian 4.19 on x64
 - C++ compiler: Unknown, distro built
 - Any non-default build flags when you build OIIO: Unknown, distro built

**Comment by lgritz** [comment]
I see, yes. The docs say:

> The `min` and `max` may either be tuples (one min and max value per channel), or single floats (same value for all channels). 

It appears that the implemented behavior when single floats are passed is to use it for only the first channel, and leave the others unclamped.

I'll fix. In the mean time, I think you can work around this by providing a tuple giving the clamp values per channel:

    b = oiio.ImageBufAlgo.clamp(a, (0,0,0), (1,1,1))

As for `clampalpha01`, what it does is 

> If `clampalpha01` is `True`, then any alpha channel is clamped to the  0--1 range.

And since you have no designated alpha channel, I'm assuming, you wouldn't expect this parameter to have any effect at all.

I'll submit a patch shortly.

**Comment by lgritz** [comment]
Fix proposed in #3265
